private/buf/cmd/buf/command/breaking/breaking.go

```diff
@@ -339,3 +351,103 @@ func validateFlags(flags *flags) error {
 	}
 	return nil
 }
+
+// hasNoUniqueBreakingConfig iterates through imageWithConfigs and checks to see if there
+// are any unique [bufconfig.BreakingConfig]. It returns true if all [bufconfig.BreakingConfig]
+// are the same across all the images.
+func hasNoUniqueBreakingConfig(imageWithConfigs []bufctl.ImageWithConfig) bool {
+	var base bufconfig.BreakingConfig
+	for _, imageWithConfig := range imageWithConfigs {
+		if base == nil {
+			base = imageWithConfig.BreakingConfig()
+			continue
+		}
+		if !equalBreakingConfig(base, imageWithConfig.BreakingConfig()) {
+			return false
+		}
+		base = imageWithConfig.BreakingConfig()
+	}
+	return true
+}
+
+// Checks if the specified [bufconfig.BreakingConfig]s are equal. Returns true if both
+// [bufconfig.BreakingConfig] have the same configuration parameters.
+func equalBreakingConfig(breakingConfig1, breakingConfig2 bufconfig.BreakingConfig) bool {
+	if breakingConfig1.Disabled() != breakingConfig2.Disabled() {
+		return false
+	}
+	if breakingConfig1.FileVersion() != breakingConfig2.FileVersion() {
+		return false
+	}
+	if slices.Compare(breakingConfig1.UseIDsAndCategories(), breakingConfig2.UseIDsAndCategories()) != 0 {
+		return false
+	}
+	if slices.Compare(breakingConfig1.ExceptIDsAndCategories(), breakingConfig2.ExceptIDsAndCategories()) != 0 {
+		return false
+	}
+	if slices.Compare(breakingConfig1.IgnorePaths(), breakingConfig2.IgnorePaths()) != 0 {
+		return false
+	}
+	for ignoreID, paths1 := range breakingConfig1.IgnoreIDOrCategoryToPaths() {
+		paths2, ok := breakingConfig2.IgnoreIDOrCategoryToPaths()[ignoreID]
+		if !ok {
+			return false
+		}
+		if slices.Compare(paths1, paths2) != 0 {
+			return false
+		}
+	}
```

Comment:
I think you could use here:
```go
maps.EqualFunc(
    breakingConfig1.IgnoreIDOrCategoryToPaths(),
    breakingConfig2.IgnoreIDOrCategoryToPaths(), 
    slices.Equal[[]string],
)
```
And then squash the if statements to lines separated by `&&` if you wanted.